src/model.cpp

```diff
@@ -14,6 +14,26 @@
 
 namespace amici {
 
+static void checkBufferSize(gsl::span<realtype> buffer,
+                            __SIZE_TYPE__ expected_size) {
+    if (buffer.size() != static_cast<size_t>(expected_size))
+        throw AmiException("Incorrect buffer size! Was %i, expected %i.",
+                           buffer.size(), expected_size);
+}
+
+static void writeSlice(std::vector<realtype> slice,
+                       gsl::span<realtype> buffer) {
+    checkBufferSize(buffer, slice.size());
+    std::copy(slice.begin(), slice.end(), buffer.data());
+}
+
+static void reset_storage(std::vector<realtype> vec, const int size) {
```

Comment:
`void std::assign (size_type n, const value_type& val);`

Should do exactly what those 4 lines do.